### PR TITLE
[UK] Update lessons/basics/modules to 1.4.0

### PR DESCRIPTION
**Pull request by fly1ngDream**
- Add newlines
- Add new content
- Fix spelling and other mistakes

### CODE REVIEW

**Comment by doomspork** [comment]
@fly1ngDream sorry for the delays. I think our @elixirschool/ukrainian team is busy with other things. Would you be so kind as to rebase your PR so I can merge this?

File: uk/lessons/basics/collections.md [hunk 1]
```diff
@@ -39,7 +40,12 @@ iex> [1, 2] ++ [3, 4, 1]
 [1, 2, 3, 4, 1]
 ```
 
-Невелика замітка щодо формату імен (`++/2`), який використовувався вище: в Elixir (та й Erlang, на основі якого створений Elixir) імена функцій та операторів складаються з двох частин: безпосередньо імені (в цьому випадку `++`) та _арності_. Арність &mdash; одне з ключових понять Elixir та Erlang. Це кількість аргументів, які приймає функція (в цьому випадку два). Арність та ім'я поєднуються через слеш. Пізніше ми розберемо це детальніше.
+Невелика замітка щодо формату імен (`++/2`), який використовувався вище: в
```
Comment (brain-geek) [inline uk/lessons/basics/collections.md]: I don't think these should be split by lines. Original is in one line.

File: uk/lessons/basics/collections.md [hunk 2]
```diff
@@ -50,14 +56,16 @@ iex> ["foo", :bar, 42] -- [42, "bar"]
...
+З лівого списку видаляється тільки перше входження кожного елемента правого списку:
 
 ```elixir
 iex> [1,2,2,3,2,3] -- [1,2,3,2]
 [2, 3]
 ```
 
-**Зауваження:** Для порівняння елементів використовується [строге порівняння](../basics#Порівняння). Наприклад:
+**Зауваження:** Для порівняння елементів використовується [строге порівняння](../basics#Порівняння). 
+Наприклад:
```
Comment (brain-geek) [inline uk/lessons/basics/collections.md]: This doesn't need a newline as well

File: uk/lessons/basics/collections.md [hunk 3]
```diff
@@ -112,9 +123,9 @@ iex> File.read("path/to/unknown/file")
 
 ## Ключові списки
 
-Ключові списки та асоціативні масиви є імплементаціями асоціативних колекцій в Elixir.
-В Elixir ключові списки &mdash; це спеціальні списки з двоелементних кортежів, першим елементом яких є атом.
-По швидкості вони ідентичні спискам.
+Ключові списки та асоціативні масиви є асоціативнними колекціми в Elixir.
+В Elixir ключові списки &mdash; це спеціальні списки з двоелементних кортежів,
```
Comment (brain-geek) [inline uk/lessons/basics/collections.md]: Please merge this into one line as well

File: uk/lessons/basics/collections.md [hunk 4]
```diff
@@ -181,11 +192,29 @@ iex> map.hello
 "world"
 ```
 
-Ще одна цікава особливість асоціативних масивів &mdash; це особливий синтаксис для оновлення (зауваження: це створює новий асоціативний масив):
+Ще одна цікава особливість асоціативних масивів &mdash; це особливий синтаксис для оновлення 
+(зауваження: це створює новий асоціативний масив):
```
Comment (brain-geek) [inline uk/lessons/basics/collections.md]: should be one line

File: uk/lessons/basics/collections.md [hunk 5]
```diff
@@ -181,11 +192,29 @@ iex> map.hello
...
 
 ```elixir
 iex> map = %{foo: "bar", hello: "world"}
 %{foo: "bar", hello: "world"}
 iex> %{map | foo: "baz"}
 %{foo: "baz", hello: "world"}
 ```
+
+**Зауваження**: цей синтаксис працює лише для оновлення ключа, який уже існує в асоціативному 
+масиві! Якщо ключ не існує, ви отримаєте `KeyError`.
```
Comment (brain-geek) [inline uk/lessons/basics/collections.md]: should be one line

File: uk/lessons/basics/modules.md [hunk 6]
```diff
@@ -1,15 +1,19 @@
 ---
-version: 1.3.0
-title: Композиція
+version: 1.4.0
+title: Модулі
 ---
 
-Ми всі знаємо з досвіду, наскільки незручно зберігати всі функції в одному файлі і в одній області видимості. В цьому уроці ми розберемося як групувати функції і визначати спеціальний асоціативний масив, відомий як `struct`, для більш ефективної організації коду.
+Ми всі знаємо з досвіду, наскільки незручно зберігати всі функції в одному файлі і в одній області
+видимості. В цьому уроці ми розберемося як групувати функції і визначати спеціальний асоціативний
```
Comment (brain-geek) [inline uk/lessons/basics/modules.md]: should be one line

File: uk/lessons/basics/modules.md [hunk 7]
```diff
@@ -55,17 +60,21 @@ defmodule Example do
...
 + `moduledoc` — Документує поточний модуль.
 + `doc` — Документація функцій і макросів.
 + `behaviour` — Використання OTP або визначеної користувачем поведінки.
 
 ## Структури
 
-Структура - спеціальний асоціативний масив з певним набором ключів і їх значеннями за замовчуванням. Вона повинна бути визначена в модулі, з якого і бере своє ім'я. Часто структура - це єдина річ, визначена в модулі.
+Структура - спеціальний асоціативний масив з певним набором ключів і їх значеннями за замовчуванням. 
+Вона повинна бути визначеною в модулі, із якого і бере своє ім'я. Часто структура - це єдина річ, 
+визначена в модулі.
```
Comment (brain-geek) [inline uk/lessons/basics/modules.md]: should be one line

File: uk/lessons/basics/modules.md [hunk 8]
```diff
@@ -55,17 +60,21 @@ defmodule Example do
...
 
-Для визначення структури ми використовуємо конструкцію `defstruct` разом з ключовим списком полів і значень за замовчуванням:
+Для визначення структури ми використовуємо конструкцію `defstruct` разом з ключовим списком полів і 
+значень за замовчуванням:
```
Comment (brain-geek) [inline uk/lessons/basics/modules.md]: should be one line

File: uk/lessons/basics/modules.md [hunk 9]
```diff
@@ -102,9 +111,44 @@ iex> %{name: "Sean"} = sean
...
+У Elixir 1.8 структури включають нестандартний самоаналіз.
+Щоб зрозуміти, що це означає і як ми повинні використовувати це, розкглянемо наш `sean` приклад:
+
+```elixir
+iex> inspect(sean)
+"#Example.User<name: \"Sean\", roles: [...], ...>"
+```
+
+Усі наші друзі присутні, і це добре для цього прикладу, але що, якщо ми мали б захищене поле, яке 
+не хотіли б включати? Нова особливість `@derive` дозволяє нам зробити це!
```
Comment (brain-geek) [inline uk/lessons/basics/modules.md]: should be one line

**Comment by tajchumber** [comment]
@fly1ngDream can you please fix the the merge conflicts before we merge please?

**Comment by brain-geek** [comment]
@fly1ngDream would it be OK for you if I fix the newlines and submit the PR once again?

**Comment by fly1ngDream** [comment]
@brain-geek yea, ofc

**Pull request closed without merging**